### ISSUE TITLE
Company completion update for recently defined symbols

### ISSUE THREAD

**Issue by 11111000000** [greghendershott/racket-mode#368]
Now I write

```
(define Тестовое-Изображение (make-object bitmap% (path->string
                      (car (Найти-Изображения
                            "/home/az/Pictures/Test")))
                                          'unknown #f #f 3))<C-x C-e>

(Тес<M-TAB>)
```
(<M-TAB> is my company-complete trigger)

and want to see completion, but it doesn't. But it works ok for symbols, defined before.

**Comment by 11111000000** [comment]
yes, maybe I can  figure out how to add any text symbols from current buffer to company-completion alist with some elisp... But why we can't update `namespace-mapped-symbols` on `racket-eval-last-sexp`?

**Comment by greghendershott** [comment]
I guess it could. But just to touch on something else, first:

Let's say you add a new definition to `file.rkt`. What next? You have a few choices, including:
1. <kbd>C-c C-c</kbd> to `racket-run`
2. <kbd>C-M-x</kbd> to `racket-eval-last-sexp`.

In general, with racket-mode, 1 is the happier path. Like DrRacket, the overall "philosophy" is: The source file module is ground truth. On every `racket-run`, the environment is _completely_ reset to be the contents of the source file module. You can explore in the REPL, by typing expressions there, as long as you want, but whatever you do there will be reset and lost on the next `racket-run`.

"Classic" lisp shortcuts like <kbd>C-M-x</kbd> exist, but effectively they're just copy and paste into the REPL.

When you're adding definitions to the source file, and want to try them out, I recommend using <kbd>C-c C-c</kbd> `racket-run`.
